src/main/java/org/jenkinsci/test/acceptance/po/PluginManager.java

```diff
@@ -76,8 +77,23 @@ public PluginManager(Jenkins jenkins) {
      * Force update the plugin update center metadata.
      */
     public void checkForUpdates() {
-        visit("checkUpdates");
-        waitFor(by.xpath("//span[@id='completionMarker' and text()='Done']"));
+        visit("advanced");
+        // The check now button is a POST with a redirect to the same page.
+        // We use the button itself to detect when the page has changed, which happens after the refresh has been done
+        final WebElement button = find(by.button("Check now"));
+        clickButton("Check now");
+        waitFor().withTimeout(30, TimeUnit.SECONDS).until(new Callable<Boolean>() {
+            @Override
+            public Boolean call() throws Exception {
+                try {
+                    button.findElement(by.id("it does not matter"));
+                } catch(StaleElementReferenceException e) {
+                    return true; // we are in another page now
+                } catch(Exception e) {
+                }
```

Comment:
does this not need a `Thread.sleep(10)` - I don't want to see a CPU core spinning, or if we are running on a single CPU VM to be fighting with the update center fetch.